frontends/PyCDE/src/pycde/support.py

```diff
@@ -22,3 +22,32 @@ def __getitem__(self, sub):
 
     raise TypeError(
         "Subscripting only supported on hw.array and hw.struct types")
+
+
+# PyCDE needs a custom version of this to support python classes.
+def var_to_attribute(obj) -> ir.Attribute:
```

Comment:
Since this function is recursive, I don't know a sane way to add the necessary support to the circt bindings version of this.

Reply:
I'm not sure I follow. If the recursion works in this version, why doesn't it work in circt.support?

Reply:
Sorry. I meant use the circt bindings version in this implementation to avoid code duplication. Since the circt bindings version (and this version) are self-recursive, I can't think of a way to call the circt bindings version to do what it can then jump back to this one.

Reply:
And I don't think we want this in the circt bindings

Reply:
got it, thanks that does make sense